lib/sync.js

```diff
@@ -360,21 +357,33 @@ SyncApi.prototype.sync = function() {
  */
 SyncApi.prototype.stop = function() {
     debuglog("SyncApi.stop");
+    if (global.document) {
+        global.document.removeEventListener("online", this._onOnline.bind(this), false);
+    }
     this._running = false;
     if (this._currentSyncRequest) { this._currentSyncRequest.abort(); }
 };
 
 /**
+ * Retry a backed off syncing request immediately. This should only be used when
+ * the user <b>explicitly</b> attempts to retry their lost connection.
+ * @return {boolean} True if this resulted in a request being retried.
+ */
+SyncApi.prototype.retryImmediately = function() {
+    if (!this._onConnectionReturned) { return false; }
+    this._startKeepAlives();
```

Comment:
won't this start another retry timer? so instead of retrying every 5 seconds or so, we'll be retrying twice every 5 seconds?